Zend/zend_language_scanner.l

```diff
@@ -2274,28 +2403,36 @@ double_quotes_scan_done:
 				}
 				/* fall through */
 			case '\n':
+				while (*YYCURSOR == ' ' || *YYCURSOR == '\t') {
+					++YYCURSOR;
+				}
+
 				/* Check for ending label on the next line */
 				if (IS_LABEL_START(*YYCURSOR) && heredoc_label->length < YYLIMIT - YYCURSOR && !memcmp(YYCURSOR, heredoc_label->label, heredoc_label->length)) {
 					YYCTYPE *end = YYCURSOR + heredoc_label->length;
 
+					if (IS_LABEL_START(*end)) {
+						continue;
+					}
+
 					if (*end == ';') {
 						end++;
 					}
 
-					if (*end == '\n' || *end == '\r') {
-						/* newline before label will be subtracted from returned text, but
-						 * yyleng/yytext will include it, for zend_highlight/strip, tokenizer, etc. */
-						if (YYCURSOR[-2] == '\r' && YYCURSOR[-1] == '\n') {
-							newline = 2; /* Windows newline */
-						} else {
-							newline = 1;
-						}
+					/* newline before label will be subtracted from returned text, but
+					 * yyleng/yytext will include it, for zend_highlight/strip, tokenizer, etc. */
+					if (YYCURSOR[-2] == '\r' && YYCURSOR[-1] == '\n') {
+						newline = 2; /* Windows newline */
+					} else {
+						newline = 1;
```

Comment:
Please no magic numbers, make a define for them.

Reply:
This is actually preexisting code (I've just removed the wrapping conditional). The numbers could be changed to macros, but this is kind of tangential to this PR.